### DISCUSSION TITLE
Any way to interpolate values in the settings.json? Trying to make asdfExecutablePath dynamic.

### DISCUSSION THREAD

**Discussion by RulerOf** [Shopify/ruby-lsp#2959] [Q&A]
I use asdf-vm to manage my Ruby versions installed on the machine. The ruby-lsp supports this, but I have to point it to the asdf installation:

```jsonc
        "rubyLsp.rubyVersionManager": {
        
            "identifier": "asdf",
            "asdfExecutablePath": "/opt/homebrew/opt/asdf/libexec/asdf.sh"
        },
```

I sync my settings between two different computers. Both Mac OS, but different architectures, so the homebrew path is different.

I'd like to do something like `"asdfExecutablePath": "$(brew --prefix asdf)/libexec/asdf.sh"`, but shelling out in a settings file seems unlikely.

Is there any solution to this that I'm possibly overlooking?

Thanks.

**Comment by andyw8** [comment]
You could try this: 

https://code.visualstudio.com/docs/editor/variables-reference#_environment-variables

**Reply by RulerOf** [reply]
I had wanted to do that, but came across [this issue in the golang extension](https://github.com/golang/vscode-go/issues/2395) that discusses how variable interpolation doesn't work in settings.json unless the extension does something to support it.

Indeed, in the doc you linked, the first paragraph calls this out:
> Visual Studio Code supports variable substitution in [Debugging](https://code.visualstudio.com/docs/editor/debugging) and [Task](https://code.visualstudio.com/docs/editor/tasks) configuration files as well as some select settings. Variable substitution is supported inside some key and value strings in launch.json and tasks.json files using ${variableName} syntax.

I decided to try it anyway, and it seems to urlencode the whole thing or something instead:
<img width="442" alt="image" src="https://github.com/user-attachments/assets/9369b3e2-80ab-453f-8deb-3f5f1b2e58f2">

I appreciate the suggestion though!

**Comment by andyw8** [comment]
Ah, yes, there are some limitations around that.

Could you perhaps use a symbolic link that it's in the same place on each machine, but pointing to different targets?

**Comment by vinistock** [comment]
@RulerOf did you try simply not specifying the `asdfExecutablePath`? Our implementation automatically checks for the ASDF installation in various places, including Homebrew on Intel architectures https://github.com/Shopify/ruby-lsp/blob/d244ad2ad450ff69343d97d88ecb2a35809e5c51/vscode/src/ruby/asdf.ts#L57.

Is that not checking in the right place?

**Reply by RulerOf** [reply]
Now *that* is surprising to me. It should have just worked without me doing anything. 

The whole thing is extra weird because it just started complaining when I used my Intel Mac under the same VS Code profile. I didn't get a sense from the code you linked if this is possibly cached some where and maybe I'm syncing it... hmm...

I'll keep this in mind as I'm gonna do some machine swapping again within the next week. I'll try to get the auto-configuration behavior working and report back. 

Thanks.
